src/ui/public/vis/vis_types/vislib_vis_type.js

```diff
@@ -23,17 +25,52 @@ export function VislibVisTypeProvider(Private) {
       this._response = esResponse;
       if (this.vis.vislibVis) {
         this.destroy();
-      } else {
-        this.vis.refreshLegend = 0;
       }
 
-      return new Promise((resolve, reject) => {
+      return new Promise(async (resolve, reject) => {
         if (!this._response) return reject();
+
+        let $scope;
+        if (this.vis.params.addLegend) {
+          const legendPositionToVisContainerClassMap = {
+            top: 'vis-container--legend-top',
+            bottom: 'vis-container--legend-bottom',
+            left: 'vis-container--legend-left',
+            right: 'vis-container--legend-right',
+          };
+
+          const getVisContainerClasses = () => {
+            return legendPositionToVisContainerClassMap[this.vis.params.legendPosition];
+          };
+
+          // update the legend class on the parent element
```

Comment:
this is bad ... (updating the parent element class name :{} ) 
however if we want to remove everything legend related from visualization for now, i don't see any other way

Reply:
I think I don;t understand the problem ;P. Why can't we require containers to have the correct legend class applied to it up front? Because it requires knowledge of the classes?

Reply:
the above comment is no longer relevant ... need to remove it

we need to update the class as the container was created in constructor, and the class might change on every render call.

a diff approach would be to create container on every render (and remove them on every render as well)
what do you think ?

Reply:
^ not a fan of recreating that container each time. I prefer your first approach, updating the class.

Reply:
Tim was against that way, as it was modifying the element that was passed to us, it could be we were breaking something

Reply:
Also we used a class name, that was not really part of vislib, but of the visualization infrastructure, and that is actually used inside `visualization` to find the container inside. I still would vote for not reapplying this class to a different element, and thus possibly mess with `visualization`, which with the current implementation of visualization shouldn't happen, but we are less error prone if we just totally ignore it.